sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1065,16 +1071,6 @@ size_t CurlSession::OnRead(uint8_t* buffer, size_t count, Context const& context
   return totalRead;
 }
 
-void CurlConnection::Shutdown()
```

Comment:
Yes - that's why I had to remove it (it was called when a Connection: close header was received).

The problem in this bug is that the `CurlConnection::Shutdown` method shut down the underlying socket, which meant that we didn't allow processing any subsequent "recv" calls, which violated the RFC.

According to the bug, the RFC says that the *only* thing we should be doing in receipt of a `Connection: close` call is to prevent the connection from being re-used, we should NOT hang up the connection.

Reply:
I guess what I am trying to understand is, with this method override gone, what is this call supposed to be doing?
https://github.com/Azure/azure-sdk-for-cpp/blob/b3b2b6774d36bfbe1d0b08174282341152af4a27/sdk/core/azure-core/src/http/curl/curl.cpp#L871-L872

Reply:
Without the virtual override, this calls the shutdown method on the base CurlConnection object.

The shutdown method on the base CurlConnection object just sets the "m_isShutdown" member variable, which tells the connection cache to not cache the connection.

Reply:
Btw, I misspoke. The virtual CurlNetworkConnection::Shutdown called ::shutdown to force the socket closed. THat was the root of the bug. My fix was to remove that force-close.

Reply:
@ahsonkhan, I am going to resolve this comment thread now so that I can merge this PR.
If you'd like more follow up your question, please create a work item so that we can follow up on it later.